app/frontend/wallet/helpers/CachedDeriveXpubFactory.ts

```diff
@@ -28,14 +37,55 @@ function CachedDeriveXpubFactory(derivationScheme, deriveXpubHardenedFn) {
     return await derivedXpubs[memoKey]
   }
 
-  async function deriveXpubNonhardenedFn(derivationPath) {
+  async function deriveXpubNonhardenedFn(derivationPath: BIP32Path) {
     const lastIndex = derivationPath.slice(-1)[0]
     const parentXpub = await deriveXpub(derivationPath.slice(0, -1))
-
     return deriveChildXpub(parentXpub, lastIndex, derivationScheme.ed25519Mode)
   }
 
-  return deriveXpub
+  function createPathBulk(derivationPath: BIP32Path): BIP32Path[] {
+    const paths: BIP32Path[] = []
+    const accountIndex = derivationPath[2] - HARDENED_THRESHOLD
+    const currentAccountPage = Math.floor(accountIndex / MAX_BULK_EXPORT_AMOUNT)
+
+    /*
+    * in case of the account 0 we append also the byron path
+    * since during byron era only the first account was used
+    */
+    if (accountIndex === 0) paths.push(BYRON_V2_PATH)
```

Comment:
in case of derivationPath being the byron account path, wouldn't this result in `BYRON_V2_PATH` being in paths twice?

Reply:
Good point, you are right, it would. Of course it cant happen right now, but in future somebody could use this in the wrong way and have a hard time debugging it. Interesting problem, if somebody asks for a byron account key before a Shelley account key this would completely break the logic of appending this key to the "first" bulk export. So the important step is to always export Shelley account key first. Currently this is ensured by the  `init()` and the `isAccountUsed()` functions which export only Shelley keys and they happen before any other pubkey export-related function.

I think we can't really enforce that Shelley pub keys will be always exported first (we should comment it somewhere in the code) but at least it makes sense to not create bulks if the path is from byron era.